src/BigIntFromString.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @since 0.5.11
+ */
+import * as t from 'io-ts'
+import { either } from 'fp-ts/lib/Either'
+import { NonEmptyString } from './NonEmptyString'
+
+/**
+ * @since 0.5.11
+ */
+export interface BigIntFromStringC extends t.Type<bigint, string, unknown> {}
+
+/**
+ * @example
+ * import { BigIntFromString } from 'io-ts-types/lib/BigIntFromString'
+ * import { right } from 'fp-ts/lib/Either'
+ * import { PathReporter } from 'io-ts/lib/PathReporter'
+ *
+ * assert.deepStrictEqual(BigIntFromString.decode('1'), right(BigInt(1)))
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('1.1')), ['Invalid value "1.1" supplied to : BigIntFromString'])
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('')), ['Invalid value "" supplied to : BigIntFromString'])
+ *
+ * @since 0.5.11
+ */
+export const BigIntFromString: BigIntFromStringC = new t.Type<bigint, string, unknown>(
+  'BigIntFromString',
+  // tslint:disable-next-line
+  (u): u is bigint => typeof u === 'bigint',
```

Comment:
You can reuse `bigint.is` here.

Reply:
This will necessitate bumping the peer dep of io-ts from 2.0.0 to 2.1.0 (according to the release notes, but that still didn't work on my end strangely) and I feel replicating that one line is worth avoiding a possible breaking change downstream

Reply:
I think the bump makes sense cause using an `io-ts-types` version with bigint support before `io-ts` had that support confuses more.

You need to modify the `tsconfig.json` to add `BigInt` support to this project: https://github.com/gcanti/io-ts/blob/master/tsconfig.json#L17

Reply:
Ok, I'll bump the `io-ts` version then. Would that be a minor or patch version for `io-ts-types`? I'll have to update the docstrings to match (still at 0.5.11 expecting a patch bump)

I found the root cause of my issue with the 2.0.0 -> 2.1.0 bump earlier: I only bumped the peerDependency section and not the devDependency. Once I bump and install io-ts 2.1.0, my tests pass without the `ESNext.BigInt` lib. But I don't know if it's wise to leave it out, so I added it

Reply:
I would also expect a patch bump cause the `io-ts` bump was just minor.

Reply:
Sounds good. Thanks for your review @mlegenhausen, please let me know if there's anything else I can help with before merging

Reply:
Just wait for the final review from @gcanti

Reply:
> I feel replicating that one line is worth avoiding a possible breaking change downstream

I agree with @EricCrosson here

Reply:
Thanks for your input @gcanti. I reverted the latest changes to the type guard and all edits to `package.json`